site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+       documented along with how many tiers of parent/child route relationships
+       are allowed
+7. A Route’s `ParentRefs` can list more than one type of parent.
+    1. Example: A `HTTPRoute`’s ParentRefs can list one (or more) `Gateways` and
+       one (or more) `HTTPRoutes` at the same time.
+
+## Binding Parent and Child Routes
+
+Binding a child route to a parent route refers to the process of reconciling
+(i.e. combining) the two routes. Binding is treated as a single transaction for
+each Parent/Child route pair. Even when a child route attaches to multiple
+`RouteRules` in the parent route, the binding between the two routes is still a
+single transaction.
+
+The binding transaction has two stages; reconciliation and activation. The
+reconciliation process produces an Ephemeral route config. After the
+reconciliation process has completed successfully, the new Ephemeral config is
+made the active Ephemeral route config and applied to the data plane.
+
+An Ephemeral route is the result of the first child route to successfully bind
+to the parent route. After that, for each child route that attempts to bind to
+that parent route, the reconciliation process is actually between the existing
+Ephemeral route config and the child route's config. This is required because,
+evaluated individually, each parent/child reconciliation may produce a valid
+config but, all of them combined may produce an invalid config.
+
+This means that, when a parent route has multiple child routes that attach to
+it, the reconciliations of the children to the parent, must be done in series
+(as opposed to in parallel).
+
+NOTE: An Ephemeral route is not a formal Kubernetes object. It is a logical
+concept used, in this document, to help describe behavior. In some cases, the
+terms "parent" and "ephemeral" route are interchangeable depending on whether or
+not an ephemeral route already exists due to a previous, successful binding of a
+child to that parent.
+
+### Failed Binding
+
+The binding transaction MUST fail if there is A non-recoverable error during the
+binding process. A failed binding should result in no changes to the
+configuration of the active Ephemeral route.
+
+A failed binding DOES NOT invalidate the parent `RouteRules` that the child
+route is trying to bind to. This is necessary because other child routes might
+bind to those parental `RouteRules` (might already be bound or may try to bind
+in the future).
+
+When a parent route's binding to a child route fails, that should not block the
+parent route's bindings with other child routes. In the same way, a child route
+failing to bind with a parent route should not block that child's bindings to
+other parent routes.
+
+## Reconciling Parent and Child Routes
+
+**A key principle guiding the design of how parent and child routes are
+combined, is that the data plane can apply the resulting route settings in a
+single processing cycle (per direction) of the network traffic.**
```

Comment:
can we add a requirement that the parent and child routes can statically (in control plane) be reduced into a single route?

Reply:
I think so. Might be nice to list it explicitly. It was a bit unclear to me since 'ephemeral route' is describing an implementation. This may be overly pedantic though

Reply:
I would extend this to: the requirement should not conflict with an implementation that reduces routes statically in an external CI/CD system, allowing the feature to work with Gateway implementations that opt out from implementing this in the control plane. 

That means the entire 'ephemeral' stuff should not be part of the spec, and the rules for resolving conflict should 
be deterministic and stateless.